tests/integration/test.changes.js

```diff
@@ -746,6 +746,86 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Immediately cancel changes', function () {
+      // fixes code coverage by ensuring the changes() listener
+      // emits 'complete' even if the db's task queue isn't
+      // ready yet
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        var changes = db.changes({live: true});
+        changes.on('error', reject);
+        changes.on('complete', resolve);
+        changes.cancel();
+      });
+    });
+
+    it('Changes with invalid ddoc view name', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: ''});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    it('Changes with invalid ddoc view name 2', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: 'a/b/c'});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    if (adapter === 'local') {
+      // This test crashes CouchDB due to
+      // https://issues.apache.org/jira/browse/COUCHDB-2765
+      it('Changes with invalid ddoc with no map function', function () {
+        var db = new PouchDB(dbs.name);
+        return db.put({
+          _id: '_design/name',
+          views: {
+            name: {
+              empty: 'sad face'
+            }
+          }
+        }).then(function () {
+          return new PouchDB.utils.Promise(function (resolve, reject) {
+            var changes = db.changes({
+              live: true,
+              filter: '_view',
+              view: 'name/name'
+            });
+            changes.on('error', resolve);
+            changes.on('change', reject);
+          });
+        });
+      });
+    }
```

Comment:
Yup, but its a bug that we do not handle this error consistently between pouchdb / couchdb, any time we have `if (adapter === 'local') {` in our tests it means we behave differently and one of the main parts of the pouchdb api is to be consistent between backends. We occasionally make an explicit choice to differ, but otherwise it should only really be plain bugs in couchdb that we expect to be fixed soon that get this. The discussion in https://issues.apache.org/jira/browse/COUCHDB-2765 suggests this is an API choice for couchdb and we should be handling it.

Ill take a look to see if we can get it consistent here given the current couch behaviour, if it takes a while them I am fine landing this and filing a follow up to remove the inconsistency but any time we are adding pouch only tests its kinda like settimeout in that its a big flag we should be thinking about how to remove